File: core/time_controller.py
class TimeController:
    def __init__(self, initial_et, time_scale=500):
        self.et = initial_et
        self.time_scale = time_scale

    def update(self, dt):
        self.et += dt * self.time_scale
        return self.et

    def set_scale(self, scale):
        self.time_scale = scale

    def get_scale(self):
        return self.time_scale

File: Assets/Scripts/Core/TimeController.cs
using UnityEngine;

public class TimeController : MonoBehaviour
{
    public double timeScale = 500;
    public double et = 0;

    void Start()
    {
        et = 2451545.0;
    }

    void Update()
    {
        et += Time.deltaTime * timeScale;
    }

    public void SetScale(double scale)
    {
        timeScale = scale;
    }
}